packages/react/src/primitives/shared/utils.ts

```diff
@@ -193,3 +196,27 @@ export const classNameModifierByFlag = (
 ): string => {
   return flag ? `${base}--${modifier}` : null;
 };
+
+export const getCSSVariableIfValueIsThemeKey = (
+  propKey: string,
+  value: string | number
+) => {
+  if (typeof value === 'number') {
+    return value;
+  }
+  const path = value.split('.');
+  let { tokens } = defaultTheme;
+  tokens = tokens[stylePropsToThemeKeys[propKey]];
+  for (let i = 0; i < path.length; i++) {
+    if (tokens) {
+      tokens = tokens[path[i]];
+      continue;
+    }
+    break;
```

Comment:
Are `continue` and `break` necessary here?

Reply:
You can run a test case to check it. Like if I have `pink.10`, it will be split into `['pink', '10']` and I want to loop through the array until the end, so `continue` is necessary cause otherwise it will break in advance. For invalid key like `invalid.xxx.xxxx`, definitely get nothing in `tokens` and so I and break the loop ahead.